src/main/java/frc/robot/subsystems/intake/IntakeIOTalonFX.java

```diff
@@ -0,0 +1,200 @@
+    leftRollerStatorCurrentStatusSignal = leftRollerMotor.getStatorCurrent();
+    drumStatorCurrentStatusSignal = drumMotor.getStatorCurrent();
+
+    rightRollerSupplyCurrentStatusSignal = rightRollerMotor.getSupplyCurrent();
+    leftRollerSupplyCurrentStatusSignal = leftRollerMotor.getSupplyCurrent();
+    drumSupplyCurrentStatusSignal = drumMotor.getSupplyCurrent();
+
+    configureIntakeRollerMotors(
+        IntakeConstants.INTAKE_RIGHT_ROLLER_MOTOR_ID, IntakeConstants.INTAKE_LEFT_ROLLER_MOTOR_ID);
+    configureIntakeDrumMotor(IntakeConstants.INTAKE_DRUM_MOTOR_ID);
+  }
+
+  @Override
+  public void updateInputs(IntakeIOInputs inputs) {
+    BaseStatusSignal.refreshAll(
+        rightRollerVelocityStatusSignal,
+        leftRollerVelocityStatusSignal,
+        drumVelocityStatusSignal,
+        rightRollerStatorCurrentStatusSignal,
+        leftRollerStatorCurrentStatusSignal,
+        drumStatorCurrentStatusSignal,
+        rightRollerSupplyCurrentStatusSignal,
+        leftRollerSupplyCurrentStatusSignal,
+        drumSupplyCurrentStatusSignal);
+
+    inputs.isRightRollerIRBlocked = rightRollerIRSensor.get();
+    inputs.isLeftRollerIRBlocked = leftRollerIRSensor.get();
+    inputs.isDrumIRBlocked = drumIRSensor.get();
+
+    inputs.rightRollerStatorCurrentAmps = rightRollerStatorCurrentStatusSignal.getValueAsDouble();
+    inputs.leftRollerStatorCurrentAmps = leftRollerStatorCurrentStatusSignal.getValueAsDouble();
+    inputs.drumStatorCurrentAmps = drumStatorCurrentStatusSignal.getValueAsDouble();
+
+    inputs.rightRollerSupplyCurrentAmps = rightRollerSupplyCurrentStatusSignal.getValueAsDouble();
+    inputs.leftRollerSupplyCurrentAmps = leftRollerSupplyCurrentStatusSignal.getValueAsDouble();
+    inputs.drumSupplyCurrentAmps = drumSupplyCurrentStatusSignal.getValueAsDouble();
+
+    inputs.rightRollerVelocityRotationsPerSecond =
+        rightRollerVelocityStatusSignal.getValueAsDouble();
+    inputs.leftRollerVelocityRotationsPerSecond = leftRollerVelocityStatusSignal.getValueAsDouble();
+    inputs.drumVelocityRotationsPerSecond = drumVelocityStatusSignal.getValueAsDouble();
+
+    inputs.rightRollerReferenceVelocityRPS = rightRollerRequestedVelocity;
+    inputs.leftRollerReferenceVelocityRPS = leftRollerRequestedVelocity;
+    inputs.drumReferenceVelocityRPS = drumRequestedVelocity;
+  }
+
+  @Override
+  public void setRightRollerVelocity(double rps) {
+    rightRollerMotor.setControl(rightRollerVelocityRequest.withVelocity(rps));
+    this.rightRollerRequestedVelocity = rps;
+  }
+
+  @Override
+  public void setLeftRollerVelocity(double rps) {
+    leftRollerMotor.setControl(leftRollerVelocityRequest.withVelocity(rps));
+    this.leftRollerRequestedVelocity = rps;
+  }
+
+  @Override
+  public void setDrumVelocity(double rps) {
+    drumMotor.setControl(drumVelocityRequest.withVelocity(rps));
+    this.drumRequestedVelocity = rps;
+  }
+
+  private void configureIntakeRollerMotors(int right_id, int left_id) {
```

Comment:
I'm not sure if two devices can share the same config objects. I would recommend creating both TalonFX objects in the constructor and passing one at a time as a parameter to this method (and invoking this method twice, once for each object).